test/rematch.jl

```diff
@@ -47,6 +47,190 @@ assertion_error = (VERSION >= v"0.7.0-DEV") ? LoadError : AssertionError
     end
 end
 
+@testset "Match using extractors" begin
+    function sub1(x) x+1 end
+
+    # sub1(4) == 3
+    let x = nothing
+        @test (@match 3 begin
+           sub1(x) => x
+        end) == 4
+
+        @test x == nothing
+    end
+
+    function cons(xs)
+        if isempty(xs)
+            nothing
+        else
+            (xs[1], xs[2:end])
+        end
+    end
```

Comment:
Please use [explicit returns for all multiline functions](https://github.com/relationalai/raistyle#method-definitions). I know its minor but we should be getting used to sticking to the style guide in our OSS as well :)

Reply:
Fixed. I also went through the entire file to fix the line lengths.